uuid.go

```diff
@@ -47,18 +47,16 @@ var (
 	poolMu      sync.Mutex
 	poolPos     = randPoolSize     // protected with poolMu
 	pool        [randPoolSize]byte // protected with poolMu
-)
-
-type invalidLengthError struct{ len int }
 
-func (err invalidLengthError) Error() string {
-	return fmt.Sprintf("invalid UUID length: %d", err.len)
-}
+	ErrInvalidUUIDFormat      = errors.New("invalid UUID format")
+	ErrInvalidURNPrefix       = errors.New("invalid urn prefix")
+	ErrInvalidBracketedFormat = errors.New("invalid bracketed UUID format")
+	ErrInvalidLength          = errors.New("invalid UUID length")
```

Comment:
This is a breaking change as it no longer include the invalid length.  There is no reason to change the underlying type of this error.  This should be

```
type LengthError struct { Length int }
```